File: app/src/main/java/com/pawels96/skyrimperkcalculator/ui/CustomDialogBuilder.java
package com.pawels96.skyrimperkcalculator.ui;

import android.app.AlertDialog;
import android.content.Context;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;

import com.pawels96.skyrimperkcalculator.R;

public class CustomDialogBuilder extends AlertDialog.Builder {

    public CustomDialogBuilder(Context context) {
        this(context, R.style.CustomDialogTheme);
    }

    public CustomDialogBuilder(Context context, int themeResId) {
        super(context, themeResId);
    }

    @Override
    public AlertDialog create() {
        AlertDialog dialog = super.create();
        if (dialog.getWindow() == null) return dialog;

        dialog.getWindow().setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
        return dialog;
    }

}

File: app/src/main/java/com/pawels96/skyrimperkcalculator/presentation/dialogs/CustomDialogBuilder.kt
package com.pawels96.skyrimperkcalculator.presentation.dialogs

import android.app.AlertDialog
import android.content.Context
import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import android.os.Build
import com.pawels96.skyrimperkcalculator.R

class CustomDialogBuilder(context: Context?, themeResId: Int = R.style.CustomDialogTheme) : AlertDialog.Builder(context, themeResId) {
    override fun create(): AlertDialog {
        val dialog = super.create()
        dialog.window?.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))

        return dialog
    }
}
